dpctl/__init__.py

```diff
@@ -111,6 +112,7 @@
     "get_current_queue",
     "get_num_activated_queues",
     "is_in_device_context",
+    "nested_context_factories",
```

Comment:
I do not see why this needs to be exposed, especially since `device_context` is on its way out. 
If it were to stay we should need to design an API to manage these factories. 

I.e. hold then in dictionary and provide functions to retrieve list of registered factories, to add a factory, to remove a factory by keys.

Reply:
@PokhodenkoSA If we can avoid adding the function here, let us do it. It just adds more clean up for future.

@oleksandr-pavlyk Given that the changes need to make it to 0.12, I am willing to let it slide. Once, `device_context` is gone a whole lot of things will have to be deprecated and removed. Basically, all the functions (`get_current_queue` and family) above `nested_context_factories`.

Reply:
> I.e. hold then in a dictionary and provide functions to retrieve the list of keys for the registered factories, to add a factory to this list with some key, to remove a factory by its key.

The factory is a key. So we can use list. Also I think we should not guaranty order of contexts.
Agree about functions.

> belong to `dpctl.utils` namespace, rather than `dpctl`

Agree.

> Once, `device_context` is gone a whole lot of things will have to be deprecated and removed.

Separate discussion needed: How to make numba redirect context active for compute follows data?

> Basically, all the functions (`get_current_queue` and family) above `nested_context_factories`.

Agree. They are located close, so clean up will be easy and in the same time.

Reply:
> If we can avoid adding the function here, let us do it.

I think list is simple enough. @oleksandr-pavlyk let me keep things as is. I know that API should be good isolation. But in this case I would like to be simple.